File: designpatterns/src/com/renho/behavior/ChainofResponsibility/ConcreteHandlerA.java
package com.renho.behavior.ChainofResponsibility;

public class ConcreteHandlerA extends Handler {

	@Override
	public void handleRequest(int request) {
		System.out.println("ConcreteHandlerA:" + request);
		if(request > 1) {
			System.out.println("无法处理,向上请求!");
			this.successor.handleRequest(request);			
		}else {
			System.out.println("ConcreteHandlerA:结束");
		}
	}

}

File: ideaworkspace/designpatterns/src/main/scala/com/renho/behavior/chainofresponsibility/ConcreteHandlerA.scala
package com.renho.behavior.chainofresponsibility

/**
  * Created by renho on 2016/7/11.
  */
class ConcreteHandlerA extends Handler {
	override def handlerRequest(request: Int): Unit = {
		if(request > 1) {
			println("无法处理, 向上请求!")
			successor.handlerRequest(request)
		} else {
			println("ConcreteHandlerA结束!")
		}
	}
}
